include/grpcpp/impl/codegen/server_interceptor.h

```diff
@@ -44,19 +45,32 @@ class ServerInterceptorFactoryInterface {
 
 class ServerRpcInfo {
  public:
+  enum class Type { UNARY, CLIENT_STREAMING, SERVER_STREAMING, BIDI_STREAMING };
```

Comment:
Don't we need `UNKNOWN` here as well?

Reply:
No, ServerRpcInfo has no default constructor because it is only allocated when the call is actually bound. This actually stems from a detail of the ServerContext: it allocates a CompletionOp which may outlast it and so it can't directly have a ServerRpcInfo object (which would want a default constructor) but rather just has a pointer to one and only allocates it when its arguments are known.

Reply:
If the goal is to eliminate UNKNOWN, then the comment here is probably not needed; the TODO comment I suggested above for ClientRpcInfo should cover this.